lib/build-utilities/migrate-initializers.js

```diff
@@ -0,0 +1,127 @@
+                         `Please refer to www.ember-fastboot.com for a migration path.`;
+    throw new Error(errorMsg);
+  }
+}
+
+/**
+ * Function to move the fastboot initializers to fastboot/app
+ * @param {Object} project
+ * @param {String} rootPath
+ */
+function _moveFastBootInitializers(project, rootPath) {
+  const deprecate = project.ui.writeDeprecateLine.bind(project.ui);
+
+  // check to see if it is a fastboot complaint addon
+  const isFastbootAddon = _checkInitializerTypeExists(rootPath, 'fastboot');
+  if (isFastbootAddon) {
+    deprecate(`Having fastboot specific code in app directory of ${rootPath} is deprecated. Please move it to fastboot/app directory.`);
+
+    const fastbootDirPath = path.join(rootPath, FASTBOOT_DIR);
+    // check if fastboot/app exists
+    if(!existsSync(fastbootDirPath)) {
+      fs.mkdirsSync(fastbootDirPath);
+    }
+
+    // copy over app/initializers/fastboot and app/instance/initializers/fastboot
+    fastbootInitializerTypes.forEach((fastbootInitializerType) => {
+      var srcFastbootPath = path.join(rootPath, 'app', fastbootInitializerType, 'fastboot');
+
+      if (existsSync(srcFastbootPath)) {
+        var destFastbootPath = path.join(fastbootDirPath, fastbootInitializerType);
+        if (!existsSync(destFastbootPath)) {
+          fs.mkdirSync(destFastbootPath);
+        }
+
+        // fastboot initializer type exists so we need to move this fastboot/app
+        var fastbootFiles = fs.readdirSync(srcFastbootPath);
+        fastbootFiles.forEach((fastbootFile) => {
+          var srcPath = path.join(srcFastbootPath, fastbootFile);
+          var destPath = path.join(destFastbootPath, fastbootFile);
+          fs.copySync(srcPath, destPath);
+
+          // delete the original path files so that there are no two initializers with the same name
+          fs.unlinkSync(srcPath);
+        });
+      }
+    });
+  }
+}
+
+/**
+ * Function that migrates the fastboot initializers for all addons.
+ *
+ * @param {Object} project
+ */
+function _migrateAddonInitializers(project) {
+  const nodeModulesPath = path.join(project.nodeModulesPath);
+
+  project.addons.forEach((addon) => {
+    const currentAddonPath = path.join(nodeModulesPath, addon.name);
+
+    _checkBrowserInitializers(currentAddonPath);
+    _moveFastBootInitializers(project, currentAddonPath);
+  });
+}
+
+/**
+ * Function to migrate fastboot initializers for host app.
+ *
+ * @param {Object} project
+ */
+function _migrateHostAppInitializers(project) {
+  const hostAppPath = path.join(project.root);
+
+  _checkBrowserInitializers(hostAppPath);
+  _moveFastBootInitializers(project, hostAppPath);
+}
+
+/**
+ * Function that migrates all addons and host app fastboot initializers to fastboot/app.
+ * It also throws an error if any addon or host app has browser forked initializers.
+ *
+ * @param {Object} project
+ */
+function migrateInitializers(project) {
```

Comment:
The breaking pre FastBoot 1.0 thing was related to the changes in [ember-useragent](https://github.com/willviles/ember-useragent), that you also reviewed. There we left the `instance-initializer/fastboot/ember-useragent` in place, so it works in pre 1.0 as well as in post 1.0 (because of the automatic migration). But because it was symlinked, and the migration works on the original files, it modified the original addon code, moving the initializer to a the new place...

So this is the point I was trying to make, that this migration did not only move files in a temporary state while building the app like a broccoli plugin would do (take the previous approach of `fastboot-filter-initializers` as an example, which filters one of the `browser|fastboot` initializers in a broccoli tree), but it modified the *original* in-repo files of the app, as well as any symlinked addon (in fact any addon in node_modules). I would expect something like a codemod to do that, but certainly not `ember s`.